content/js/users.md

```diff
@@ -1,40 +1,59 @@
 <a name="manage-user"></a>
 ## Manage User
 
+### How does it work
+
+Notice that even without logging in, user can still interact with the SDK,
+such as querying data in `skygear.publicDB`. However, this behavior can be
+changed by setting different access control for public. For more information,
+please refer to [access control](/js/guide/access-control#acl-default) page.
+Skygear uses `accessToken` with localstorage to track user authentication.
+Typical flow is shown below:
+
+1. Web page load/refresh
+2. SDK checks if user information exists in localstorage
+  - yes: SDK checks if `accessToken` has expired
+    + yes: user has to log in again
+    + no: user remains logged in (skip 3)
+  - no: user has to log in or sign up
+3. User logs in or signs up (trigger `onUserChanged`)
+4. SDK updates user information in container and localstorage
+  - `skygear.currentUser` is now available
+  - `skygear.accessToken` is now available
+5. User logs out or `accessToken` expires (trigger `onUserChanged`)
+6. SDK clears user information in container and localstorage
+
 ### Sign up
 
-If you are not familiar with Promises, please read more [here](https://www.promisejs.org/).
+To deal with duplicate username or email error, please refer to
+[handling errors](/js/guide/handling-errors#error-code) page.
```

Comment:
Not sure if it is good to redirect the reader to the "Handling Errors" page for this. We can discuss this further